src/escape.rs

```diff
@@ -131,6 +155,116 @@ where
         }
         i += 1;
     }
-    w.write_all(&bytes[mark..])?;
+    w.write_all(&bytes[mark..])
+}
+
+/// Calls the given function with the index of every byte that is in
+/// [", &, <, >] and for no other byte.
+#[cfg(all(target_arch = "x86_64", feature="simd"))]
+fn scan_simd<F>(bytes: &[u8], mut offset: usize, mut callback: F) -> io::Result<()>
+    where F: FnMut(usize) -> io::Result<()>
+{
+    if bytes.is_empty() {
+        // we need this special case for when we are passed an empty subslice
+        // that is exactly on a page boundary. we would try to read on a page
+        // that is potentially not ours and crash
+        return Ok(());
+    }
+
+    let lower_vec = unsafe { _mm_set_epi8( 0, 62, 0, 60, 0, 0, 0, 0, 0, 38, 0, 0, 0, 34, 0, 127 ) };
+    let upperbound = bytes.len().saturating_sub(16);
+
+    while offset < upperbound {
+        let mut mask = unsafe {
+            let raw_ptr = transmute(bytes.as_ptr().offset(offset as isize));
```

Comment:
You don't need a transmute here. See the `memchr` crate for how to use pointer casts instead.

Reply:
:+1: